### PR TITLE
Adds Conversion from RST Types to TST Kinds

### PR DESCRIPTION
**Pull request by MarcoTz**
The kindInference module looks up the right kinds for each RST Type and writes them into the resulting TST Kind

### CODE REVIEW

**Review by skingo — APPROVED** [review]

File: src/Driver/Driver.hs [hunk 1]
```diff
@@ -203,32 +205,48 @@ inferDecl :: ModuleName
...
   -- Insert into environment
-  let f env = env { declEnv = (RST.data_loc decl,decl) : declEnv env }
+  let f env = env { declEnv = (RST.data_loc decl,decl) : declEnv env, kindEnv = insertKinds decl (kindEnv env)}
+
   modifyEnvironment mn f
   pure (TST.DataDecl decl)
+  where 
+    insertKinds :: RST.DataDecl -> Map XtorName MonoKind -> Map XtorName MonoKind
+    insertKinds RST.NominalDecl{data_kind = knd, data_xtors = xtors} mp = do
+      let names = map RST.sig_name (fst xtors) ++ map RST.sig_name (snd xtors)
```
Comment (BinderDavid) [inline src/Driver/Driver.hs:229]: Don't insert twice (remove everything after `++`).

File: src/Dualize/Terms.hs [hunk 2]
```diff
@@ -44,7 +44,7 @@ dualCmd :: Command -> Either DualizeError Command
 dualCmd (Apply _ annot kind prd cns) = do
     t1 <- dualTerm CnsRep cns
     t2 <- dualTerm PrdRep prd
-    return $ Apply defaultLoc (dualApplyAnnot annot) (dualMonoKind kind) t1 t2
+    return $ Apply defaultLoc (dualApplyAnnot annot) kind t1 t2--(dualMonoKind kind) t1 t2
```
Comment (BinderDavid) [inline src/Dualize/Terms.hs:47]: reinsert dualMonoKind

File: src/TypeAutomata/FromAutomaton.hs [hunk 3]
```diff
@@ -27,6 +27,11 @@ import Data.Graph.Inductive.Graph
 import Data.Graph.Inductive.Query.DFS (dfs)
 import Data.List.NonEmpty (NonEmpty)
 
+-- default kind
+-- will be changed to actual kind later, once type automatat have been adjuste
```
Comment (BinderDavid) [inline src/TypeAutomata/FromAutomaton.hs:31]: Add deprecated

Comment (BinderDavid) [reply src/TypeAutomata/FromAutomaton.hs:31]: Doesn't warn within the defining module :(

File: src/Dualize/Terms.hs (hunk 2 above)
Comment (BinderDavid) [reply src/Dualize/Terms.hs:47]: Done

**Pull request merged**